live-ingester/src/main/java/org/alfresco/hxi_connector/live_ingester/adapters/messaging/repository/mapper/CamelEventMapper.java

```diff
@@ -58,4 +65,29 @@ public RepoEvent<DataAttributes<NodeResource>> repoEventFrom(Exchange exchange)
             throw new LiveIngesterRuntimeException("Event deserialization failed", e);
         }
     }
+
+    /**
+     * This method alters the original type of repo event.
+     *
+     * @param exchange
+     *            Camel exchange object
+     * @param newEventType
+     *            event type to be altered
+     * @return altered repo event
+     */
+    public RepoEvent<DataAttributes<NodeResource>> alterRepoEvent(Exchange exchange, String newEventType)
```

Comment:
This method could take `RepoEvent` as param instead of whole `Exchange`, WDYT?

Reply:
It could, however, to me, it is neater to use `Exchange` object.
Look at the usage of this method - there would be unnecessary boilerplate around it to get the repo event from `Exchange` and set it back as body.
Actually, when you raised this comment - I thought this method could be void which would handle all needed changes inside `Exchange` object so that it calling it would be simpler. However, this might not fit to `CamelEventMapper` class responsibilities (current method is already not quite fitting).
WDYT?

Reply:
I would leave it as is - I agree that `void` won't fit into this class